rust/lance/src/dataset/scanner.rs

```diff
@@ -522,15 +573,101 @@ impl Scanner {
             extra_columns.push(vector_field);
             extra_columns.push(ArrowField::new(DIST_COL, DataType::Float32, true));
         };
+
         if self.with_row_id {
             extra_columns.push(ROW_ID_FIELD.clone());
         }
 
         let schema = if !extra_columns.is_empty() {
-            self.projections.merge(&ArrowSchema::new(extra_columns))?
+            self.phyical_columns
+                .merge(&ArrowSchema::new(extra_columns.clone()))?
         } else {
-            self.projections.clone()
+            self.phyical_columns.clone()
         };
+
+        // drop metadata
+        // NOTE: this is the current behavior as we don't return metadata in queries
+        // We should reconsider this behavior
+        if self.nearest.is_some() {
+            Ok(Arc::new(Schema {
+                fields: schema.fields,
+                metadata: HashMap::new(),
+            }))
+        } else {
+            Ok(Arc::new(schema))
+        }
+    }
+
+    pub(crate) fn output_expr(&self) -> Result<Vec<(Arc<dyn PhysicalExpr>, String)>> {
```

Comment:
It's a little weird to me that there is both a variable named `output_expr` and a method named `output_expr` and that method isn't just returning the variable.  Can you rename one of them?

Reply:
changed the variable to `requested_output_expr` to denote that it is the input from the user.